scripts/ci/jobs/diff-dumps.sh

```diff
@@ -111,22 +111,43 @@ upload_offline_dump() {
     info "Uploading offline dump"
 
     cd /tmp/offline-dump
-    cmd=()
-    if is_in_PR_context; then
-        cmd+=(echo "Would do")
-    fi
+
     curl --silent --show-error --max-time 60 --retry 3 --create-dirs -o out/RELEASE_VERSION.txt https://raw.githubusercontent.com/stackrox/stackrox/master/scanner/updater/version/RELEASE_VERSION
     version_file="out/RELEASE_VERSION.txt"
-    # Use grep to extract X.Y versions, sort them, and get the last one as the latest
-    latest_version=$(grep -oE '^[0-9]+\.[0-9]+' "$version_file" | sort -V | tail -n 1)
+
+    latest_version=""
+    # Get the latest X.Y version which has been released.
+    while read -r version; do
+        status_code=$(curl --silent --max-time 60 --retry 3 --write-out "%{http_code}" -o /dev/null "https://github.com/stackrox/stackrox/releases/tag/$version.0")
+        if [[ "$status_code" == "404" ]]; then
+            # Release was not cut, so try a previous release.
+            continue
+        elif [[ "$status_code" == "200" ]]; then
+            # Release was cut, so use this one.
+            latest_version="$version"
+            break
+        else
+            # Some other error occurred.
+            >&2 echo "received status code $status_code when fetching StackRox releases"
+            exit 1
+        fi
+    done < <(grep -oE '^[0-9]+\.[0-9]+' "$version_file" | sort -ruV)
```

Comment:
Note I updated the flags here to `-ruV` instead of just `-V`. This way we can try the latest version first (`-r`). Also no need to try an X.Y release more than once, hence the `-u`

Reply:
I find the long-hand name for flags more helpful when reading bash; do you mind changing them here?

Reply:
yep, done [276e3a8](https://github.com/stackrox/scanner/pull/1568/commits/276e3a8ecd39e51216d374a9cee73b1a65dbe20a)